amundsen_application/static/js/components/DashboardPage/index.tsx

```diff
@@ -0,0 +1,264 @@
+import { GetDashboardRequest } from 'ducks/dashboard/types';
+import { GlobalState } from 'ducks/rootReducer';
+import { logClick } from 'ducks/utilMethods';
+import { Dashboard } from 'interfaces/Dashboard';
+import ImagePreview from './ImagePreview';
+import QueryList from 'components/DashboardPage/QueryList';
+import ChartList from 'components/DashboardPage/ChartList';
+import { formatDateTimeShort } from '../../utils/dateUtils';
+import ResourceList from 'components/common/ResourceList';
+import {
+  ADD_DESC_TEXT,
+  DASHBOARD_OWNER_SOURCE,
+  DASHBOARD_SOURCE,
+  LAST_RUN_SUCCEEDED,
+  NO_OWNER_TEXT,
+  TABLES_PER_PAGE
+} from 'components/DashboardPage/constants';
+import TagInput from 'components/Tags/TagInput';
+import { EditableSection } from 'components/TableDetail/EditableSection';
+import { ResourceType } from 'interfaces';
+
+import { getSourceDisplayName, getSourceIconClass } from 'config/config-utils';
+
+import './styles.scss';
+
+export interface RouteProps {
+  uri: string;
+}
+
+interface DashboardPageState {
+  uri: string;
+}
+
+export interface StateFromProps {
+  isLoading: boolean;
+  statusCode: number;
+  dashboard: Dashboard;
+}
+
+export interface DispatchFromProps {
+  getDashboard: (payload: { uri: string, index?: string, source?: string }) => GetDashboardRequest;
+}
+
+export type DashboardPageProps = RouteComponentProps<RouteProps> & StateFromProps & DispatchFromProps;
+
+export class DashboardPage extends React.Component<DashboardPageProps, DashboardPageState> {
+  constructor(props) {
+    super(props);
+
+    const { uri } = qs.parse(this.props.location.search);
+    this.state = { uri };
+  }
+
+  componentDidMount() {
+    this.loadDashboard();
+  }
+
+  loadDashboard() {
+    this.props.getDashboard({ uri: this.state.uri });
+  }
+
+  mapStatusToStyle = (status: string): string => {
+    if (status === LAST_RUN_SUCCEEDED) {
+      return 'success';
+    }
+    return 'danger';
+  };
+
+  renderTabs() {
+    const tabInfo = [];
+    tabInfo.push({
+      content:
+        (
+          <ResourceList
+            allItems={ this.props.dashboard.tables }
+            itemsPerPage={ TABLES_PER_PAGE }
+            paginate={ false }
+            source={ DASHBOARD_SOURCE }
+          />
+        ),
+      key: 'tables',
+      title: `Tables (${this.props.dashboard.tables.length})`,
+    });
+
+    if (this.props.dashboard.chart_names.length > 0 ) {
+      tabInfo.push({
+        content: <ChartList charts={ this.props.dashboard.chart_names }/>,
+        key: 'charts',
+        title: `Charts (${this.props.dashboard.chart_names.length})`,
+      });
+    };
+
+    tabInfo.push({
+      content: <QueryList queries={ this.props.dashboard.query_names }/>,
+      key: 'queries',
+      title: `Queries (${this.props.dashboard.query_names.length})`,
+    });
+
+    return <TabsComponent tabs={ tabInfo } defaultTab={ "tables" } />;
+  }
+
+  render() {
+    const { dashboard, isLoading } = this.props;
```

Comment:
nit: for tests check if at minimum any boolean cases are covered and verify certain components show up with the expected props.